### ISSUE TITLE
CLI Decoding Error

### ISSUE THREAD

**Issue by chayim** [RedisJSON/RedisJSON#737]
I created a JSON object like the following using json.set (json.set foo .)

"{\"hello\":\"world\",\"versions\":[{\"my\":\"potato\",\"likes\":\"your potato\"},{\"my\":\"butter\",\"likes\":\"cheese\"}]}"

Issuing the following commands in the redis-cli yields a decoding error:
```
127.0.0.1:6379> JSON.GET foo $.versions[*][?@(my.potato)]
(error) JSON Path error: path error: \n$.versions[*][?@(my.potato)]\n^^^^^^^^^^^^^^^^^\n
```

Simiilary, it's reproducible with another related query:

```
127.0.0.1:6379> JSON.GET foo $.versions[*][?@(my=potato)]
(error) JSON Path error: path error: \n$.versions[*][?@(my=potato)]\n^^^^^^^^^^^^^^^^^\n
```

**Comment by chayim** [comment]
Ignore - I replaced your key with mine. BUT what about the cli decoding issue?

**Comment by oshadmi** [comment]
You can fix it to be legal (inside the parenthesis, dot following the `@`, double equals and quotes around value, all the path expression wrapped with single quote)
```
127.0.0.1:6379> JSON.GET foo '$.versions[*][?(@.my=="potato")]'
```

**Comment by chayim** [comment]
Yup - did this today with quoting. But, we should fix the CLI issue.
